src/api/flash.js

```diff
@@ -59,80 +60,67 @@ export const flash = async (flasher, board, port, filename, options) => {
       };
 
   /***
-   * If already in bootloader mode, flash away and we're done.
-   ***/
-  if (device.bootloader) {
-    return await flasher.flash(board, port, filename, options);
-  }
-
-  /***
-   * Preparations before flashing:
-   * - If doing a factory reset, do a factory reset.
-   * - If not, save a structured eeprom backup.
-   ***/
+   * If we're already in bootloader mode, skip the preparations.
+   */
   let saveKey;
-  if (!options.factoryReset) {
-    await toStep(callback)("saveEEPROM");
-    saveKey = await focusCommands.saveEEPROM();
-  } else {
-    await toStep(callback)("factoryRestore");
-  }
-
-  // Clear the EEPROM after saving. We do this so that if the layout changed,
-  // any space that changed owners will not have garbage in it, but either
-  // uninitialized bytes, or whatever the restore later restores.
-  try {
-    await focusCommands.eraseEEPROM();
-  } catch (e) {
-    if (e != "Communication timeout") {
-      throw new Error(e);
+  if (!startFromBoot) {
+    /***
+     * Preparations before flashing:
+     * - If not doing a factory reset, save a structured eeprom backup.
+     * - Otherwise proceed.
+     ***/
+    if (!options.factoryReset) {
+      await toStep(callback)("saveEEPROM");
```

Comment:
The whole 'toStep' thing would be clearer if we called it something like 'reportUpdateStatus'